api/buildings/views.py

```diff
@@ -59,32 +62,40 @@ def public_create(self, request):
                 serializer.errors, status=status.HTTP_400_BAD_REQUEST
             )
 
-
-@api_view(["GET"])
-@permission_classes((permissions.AllowAny,))
-@extend_schema(
-    parameters=[
-        BuildingSearchSerializer,
-        OpenApiParameter(
-            "query", BuildingSearchSerializer, OpenApiParameter.QUERY
-        ),
-    ],
-    description="Some cool building search",
-)
-def building_search(request):
-    query = request.GET.get("query")
-
-    buildings = (
-        Building.approved.annotate(
-            similarity=TrigramSimilarity("address", query)
-        )
-        .filter(similarity__gt=settings.TRIGRAM_SIMILARITY_THRESHOLD)
-        .order_by("-similarity")
+    @extend_schema(
+        parameters=[
+            OpenApiParameter("query", SearchQuerySerializer, OpenApiParameter.QUERY)
```

Comment:
Would add `description="The address of the building"` as well to be more specific as to what the input is